pyproject.toml

```diff
@@ -1,63 +1,79 @@
-[tool.poetry]
+[build-system]
+requires = [
+    "setuptools >= 40.0.4",
+    "wheel >= 0.29.0",
+    "setuptools-scm>=8",
+]
+build-backend = 'setuptools.build_meta'
+
+[project]
 name = "argus-alm"
-version = "0.15.0"
 description = "Argus"
-authors = ["Alexey Kartashov <alexey.kartashov@scylladb.com>", "Łukasz Sójka <lukasz.sojka@scylladb.com>"]
+authors = [
+    {name = "Alexey Kartashov", email = "alexey.kartashov@scylladb.com"},
+    {name = "Łukasz Sójka", email = "lukasz.sojka@scylladb.com"},
+]
 license = "Apache-2.0"
-repository = "https://github.com/scylladb/argus"
 readme = "README.md"
-packages = [
-    { include = "argus/client" },
-    { include = "argus/common" },
-    { include = "argus/__init__.py" },
+requires-python = ">=3.10"
+dependencies = [
+    "requests >= 2.26.0",
+    "click >= 8.1.3"
 ]
-exclude = ["argus/client/tests"]
+dynamic = ["version"]
 
-[tool.poetry.group.web-backend]
-optional = true
+[project.urls]
+Homepage = "https://github.com/scylladb/argus"
+Repository = "https://github.com/scylladb/argus"
 
-[tool.poetry.group.docker-image]
-optional = true
 
-[tool.poetry.dependencies]
-requests = "^2.26.0"
-python = "^3.10"
-click = "^8.1.3"
-
-[tool.poetry.scripts]
-argus-client-generic = 'argus.client.generic.cli:cli'
-argus-driver-matrix-client = 'argus.client.driver_matrix_tests.cli:cli'
+[tool.setuptools_scm]
+version_file = "argus/_version.py"
 
-[tool.poetry.group.web-backend.dependencies]
-PyYAML = "^6.0.0"
-scylla-driver = "^3.26.8"
-Flask = "^3.0.0"
-Flask-WTF = "^1.0.0"
-Flask-Login = "^0.5.0"
-humanize = "^3.13.1"
-python-magic = "^0.4.24"
-uwsgi = "^2.0.20"
-python-jenkins = "^1.7.0"
-python-slugify = "^6.1.1"
-pygithub = "^2.6.1"
-boto3 = "^1.38.9"
+[tool.setuptools.packages.find]
+include = ["argus", "argus.client*", "argus.common*"]
+# exclude = ["backend*"]
 
-[tool.poetry.group.docker-image.dependencies]
-supervisor = "^4.2.4"
+# [tool.setuptools.exclude-package-data]
+# "argus" = ["backend*", "backend.**"]
 
-[tool.poetry.group.dev.dependencies]
-boto3-stubs = {extras = ["s3"], version = "^1.38.9"}
-coverage = "5.5"
-docker = "7.1.0"
-pytest = "6.2.5"
-ipython = "^8.1"
-pre-commit = "^4.2.0"
-ruff = "^0.11.9"
-autopep8 = "^2.3.2"
+[project.scripts]
+argus-client-generic = 'argus.client.generic.cli:cli'
+argus-driver-matrix-client = 'argus.client.driver_matrix_tests.cli:cli'
 
-[build-system]
-requires = ["poetry-core>=1.0.0"]
-build-backend = "poetry.core.masonry.api"
+[project.optional-dependencies]
```

Comment:
Any option in UV to keep using semver? AFAIK `>=` for, example, `3.10.0` will pull `4.0.0` as a valid installation compared to `^3.10.0` which will always be limited to `3.x.y`

Reply:
Should be safe to update